File: source_funcs/go_transcoder/MINIMUM_XOR_VALUE_PAIR_1/MINIMUM_XOR_VALUE_PAIR_1.go
package main

import (
	"math"
	"sort"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}

func f_gold(arr []int, n int) int {
	sort.Ints(arr)
	var minXor int = math.MaxInt64
	var val int = 0
	for i := int(0); i < n-1; i++ {
		val = arr[i] ^ arr[i+1]
		minXor = min(minXor, val)
	}
	return minXor
}

File: rust_funcs/cpp_transcoder/MINIMUM_XOR_VALUE_PAIR_1/MINIMUM_XOR_VALUE_PAIR_1.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::f32;

fn f_gold(arr: &mut [f32], n: i32) -> i32 {
    arr.sort_by(|a, b| a.partial_cmp(b).unwrap());
    let mut min_xor = i32::MAX; 
    let mut val = 0;
    for i in 0..n-1 {
        val = (arr[i] as i32) ^ (arr[i + 1] as i32);
        min_xor = min(min_xor, val);
    }
    min_xor
}
